pkg/agent/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -110,6 +111,8 @@ func NewNetworkPolicyController(antreaClientGetter agent.AntreaClientProvider,
 	ifaceStore interfacestore.InterfaceStore,
 	nodeName string,
 	entityUpdates <-chan types.EntityReference,
+	groupCounters []types2.GroupCounter,
```

Comment:
I feel like the NetworkPolicyController doesn't really need to be able to access the `GroupCounter` objects. If we get all the updates in the channel (svcName + group ID(s)), then we could store that data locally as well. Maybe @tnqn has an opinion here, or an alternative approach to suggest.

Reply:
If we store the data locally(in agent NP controller) for example in a map, then this map, IMO, also needs to add a lock to make it thread-safe. Since the groupCounter has a thread-safe mechanism, is it good that we could reuse it? Glad to hear your ideas?